core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -403,8 +403,12 @@ private[log] class Cleaner(val id: Int,
    * @param retainDeletes Should delete tombstones be retained while cleaning this segment
    *
    */
-  private[log] def cleanInto(topicAndPartition: TopicAndPartition, source: LogSegment,
-                             dest: LogSegment, map: OffsetMap, retainDeletes: Boolean) {
+  private[log] def cleanInto(topicAndPartition: TopicAndPartition,
+                             source: LogSegment,
+                             dest: LogSegment,
+                             map: OffsetMap,
+                             retainDeletes: Boolean,
+                             messageFormatVersion: Int) {
```

Comment:
I had a comment on this in the previous round of review. It seems that during compaction, it would be better to write the message in the configured message format : (1) This reduces the message conversion during fetch. (2) Converting to message format V1 allows us to track timestamp at the message level, which will be useful for things like removing the tombstone.